haystack/preview/utils/filters.py

```diff
@@ -48,7 +48,7 @@ def or_operation(conditions: List[Any], document: Document, _current_key: str):
     :return: True if the document matches ano of the filters, False otherwise
     """
     for condition in conditions:
```

Comment:
This could be:
`return any(document_matches_filter(conditions=condition, document=document, _current_key=_current_key) for condition in conditions)`